exchanges/bittrex/bittrex_websocket.go

```diff
@@ -0,0 +1,618 @@
+package bittrex
+
+import (
+	"bytes"
+	"compress/flate"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/gorilla/websocket"
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/common/crypto"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/stream"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+const (
+	bittrexAPIWSURL             = "wss://socket-v3.bittrex.com/signalr"
+	bittrexAPIWSNegotiationsURL = "https://socket-v3.bittrex.com/signalr"
+
+	bittrexWebsocketTimer = 13 * time.Second
+	wsTicker              = "ticker"
+	wsOrderbook           = "orderbook"
+	wsMarketSummary       = "market_summary"
+	wsOrders              = "order"
+	wsHeartbeat           = "heartbeat"
+	authenticate          = "Authenticate"
+	subscribe             = "subscribe"
+	unsubscribe           = "unsubscribe"
+	wsRateLimit           = 50
+	wsMessageRateLimit    = 60
+)
+
+var defaultSpotSubscribedChannels = []string{
+	// wsHeartbeat,
```

Comment:
The Bittrex websocket engine also sends out a ping, and since we don't use the heartbeat, it's mostly redundant with this ping.
Although there's a tiny difference: the heartbeat sends a complete packet inside the signalR protocol, and that demonstrates that a bigger part of the stack functions fine than the ping. So that's why I want to use it locally so send empty updates to the orderbook to prevent the timeout on illiquid books.
However, I don't feel strongly about it: removing, enabling, disabling or commenting out is all fine with me.